metrics/metrics-integration/metrics-prometheus/src/main/java/org/apache/servicecomb/metrics/prometheus/MetricsHttpPublisher.java

```diff
@@ -37,20 +36,34 @@
 public class MetricsHttpPublisher implements ApplicationListener<ApplicationEvent> {
   private static final Logger LOGGER = LoggerFactory.getLogger(MetricsHttpPublisher.class);
 
-  private static final String METRICS_PROMETHEUS_PORT = "servicecomb.metrics.prometheus.port";
+  private static final String METRICS_PROMETHEUS_ADDRESS = "servicecomb.metrics.prometheus.address";
 
   private HTTPServer httpServer;
 
   public MetricsHttpPublisher() {
     //prometheus default port allocation is here : https://github.com/prometheus/prometheus/wiki/Default-port-allocations
-    int publishPort = DynamicPropertyFactory.getInstance().getIntProperty(METRICS_PROMETHEUS_PORT, 9696).get();
-    MetricsCollector metricsCollector = new MetricsCollector();
-    metricsCollector.register();
-    try {
-      this.httpServer = new HTTPServer(new InetSocketAddress(publishPort), CollectorRegistry.defaultRegistry, true);
-      LOGGER.info("Prometheus httpServer listened {}.", publishPort);
-    } catch (IOException e) {
-      throw new ServiceCombException("create http publish server failed", e);
+    this.init(
+        DynamicPropertyFactory.getInstance().getStringProperty(METRICS_PROMETHEUS_ADDRESS, "localhost:9696").get());
```

Comment:
1.need to care for compatible?
2.is it better to be: 0.0.0.0:9696?

Reply:
I think this address is for the Prometheus client to connect the server.

Reply:
Yes,Willem is right

Reply:
change to "0.0.0.0:9696" as default